Sources/MongoSwift/SDAM.swift

```diff
@@ -223,6 +223,29 @@ public struct ServerDescription {
         } ?? []
         self.tags = hello?.tags ?? [:]
     }
+
+    internal init(type: ServerType
```

Comment:
I wasnt really able to do this easily. I tried just copy and pasting the initializers into extensions but that did not work, I got a warning suggesting I reformat the initializer to do `self.init()` but once I did that it caused a lot of other errors. Is there anything you suggest? For now, I added comments on the initializers saying they are for testing purposes

Reply:
iirc, initializers defined in extensions in Swift have to delegate to initializers in the type's initial definition, I think that's what you're running into here. this is a little restrictive but I think makes sense; without the restriction it would be easy for a driver user to extend a type and add their own initializer that e.g. bypasses validation of some property, and results in their creating an invalid object.

the only existing initializer takes in a libmongoc object so delegating to it won't work.

one option would be to add a new initializer here that takes in all of the stored properties as parameters (maybe would be useful in the future if/when we need to create ServerTypes from elsewhere?), and then you could add via extension in the tests some static method like `makeTestServerDescription(serverType:)` that delegates to that `init` and fills in the default values you use here.

on the other hand, once we have the spec tests running maybe we don't need the handwritten tests you're adding anymore. for the spec tests I think we might just be creating these `ServerDescription`s by decoding them from extJSON, so maybe this initializer would not even be needed by the time the project is finished?  if so, I think the approach of just noting in a comment that it's for testing is good enough